src/modules/navigator/rtl.cpp

```diff
@@ -43,43 +43,71 @@
 #include "navigator.h"
 #include <dataman/dataman.h>
 
-
 static constexpr float DELAY_SIGMA = 0.01f;
 
+using namespace time_literals;
+
 RTL::RTL(Navigator *navigator) :
 	MissionBlock(navigator),
 	ModuleParams(navigator)
 {
 }
 
-void
-RTL::on_inactivation()
+void RTL::on_inactivation()
 {
 	if (_navigator->get_precland()->is_activated()) {
 		_navigator->get_precland()->on_inactivation();
 	}
 }
 
-void
-RTL::on_inactive()
+void RTL::on_inactive()
 {
 	// Reset RTL state.
 	_rtl_state = RTL_STATE_NONE;
 
 	find_RTL_destination();
-
 }
 
-void
-RTL::find_RTL_destination()
+void RTL::find_RTL_destination()
 {
-	// get home position:
-	home_position_s &home_landing_position = *_navigator->get_home_position();
-	// get global position
-	const vehicle_global_position_s &global_position = *_navigator->get_global_position();
+	// require updated global position
+	vehicle_global_position_s global_position;
+
+	if (!_vehicle_global_position_sub.update(&global_position)) {
+		return;
+	}
```

Comment:
We should not update it again here, we already do that in the main while loop:
https://github.com/PX4/Firmware/blob/a9dacae57e1a20636ffb7d386450790725f1c0f6/src/modules/navigator/navigator_main.cpp#L196-L203

Reply:
Yes it seems redundant, but the copy is cheap (1 or 2 microseconds) and it provides a mechanism to skip running the expensive RTL destination find (with dataman reads) when global position isn't available for whatever reason.

Reply:
But it won't ever be updated because we update it straight before running this code. It doesn't make much senses to me.